### PR TITLE
[FLINK-34126][configuration] Correct the description of jobmanager.scheduler

### PR DESCRIPTION
**Pull request by JunRuiLee**
## What is the purpose of the change

More details see: https://issues.apache.org/jira/browse/FLINK-34126

## Brief change log

Correct the description of jobmanager.scheduler

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by JunRuiLee** [comment]
@flinkbot run azure

**Comment by JunRuiLee** [comment]
@flinkbot run azure

**Comment by JunRuiLee** [comment]
@flinkbot run azure

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 1]
```diff
@@ -458,29 +460,46 @@ public enum JobStoreType {
...
     /** Type of scheduler implementation. */
-    public enum SchedulerType {
+    public enum SchedulerType implements DescribedEnum {
         /** @deprecated Use {@link SchedulerType#Default} instead. */
         @Deprecated
-        Ng,
-        Default,
-        Adaptive,
-        AdaptiveBatch
+        Ng(),
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:473]: It's better to describe it as "Deprecated. Use Default scheduler instead."

Then we can remove the paramless ctor.

**Review by zhuzhurk — APPROVED** [review]
LGTM.

**Pull request closed without merging**